src/pilot/assign_role.py

```diff
@@ -980,18 +994,39 @@ def generate_osd_config(ip_mac_service_tag, drac_client):
 
     # Using the simple yaml.dump results in a completely unreadable file, so we
     # do it the hard way to create something more user friendly
-    with open(osd_config_file + ".orig", 'r') as instream:
-        with open(osd_config_file, 'w') as outstream:
-            for line in instream:
-                if '{}' not in line:
-                    outstream.write(line)
+    instream = open(osd_config_file + ".orig", 'r')
+    while True:
+        try:
+            fcntl.flock(instream, fcntl.LOCK_EX | fcntl.LOCK_NB)
```

Comment:
I don't think it's necessary to lock on the .orig file.  This file is shipped as part of JetPack and it's never changed.  It should be ok to have multiple threads reading this file at once.